plugin/src/test/groovy/com/novoda/buildproperties/internal/FilePropertiesEntriesTest.groovy

```diff
@@ -44,12 +47,12 @@ class FilePropertiesEntriesTest {
     }
 
     @Test
-    void shouldThrowIllegalArgumentExceptionWhenTryingToAccessValueOfUndefinedProperty() {
+    void shouldThrowExceptionWhenTryingToAccessValueOfUndefinedProperty() {
         try {
             entries['notThere'].string
-            fail('IllegalArgumentException expected')
-        } catch (IllegalArgumentException e) {
-            assertThat(e.getMessage()).startsWith("No value defined for property 'notThere'")
+            fail('Exception expected')
+        } catch (Exception e) {
+            assertThat(e.getMessage()).contains("Unable to find value for key 'notThere'")
```

Comment:
I considered `ExpectedException` when wrote these tests the first time, but the thing I dislike from that rule is you have to write your assertions before the exception is thrown, which basically breaks the _given/when/then_ structure I like to keep in my tests.